server/jamovi/core/columnw.cpp

```diff
@@ -117,31 +117,59 @@ void ColumnW::appendLevel(int value, const char *label)
 
     int length = strlen(label)+1;
     size_t allocated;
-
     char *chars = _mm->allocate<char>(length, &allocated);
-
     std::memcpy(chars, label, length);
 
+    length = strlen(importValue)+1;
+    size_t importAllocated;
+    char *importChars = _mm->allocate<char>(length, &importAllocated);
+    std::memcpy(importChars, importValue, length);
+
     s = struc();
     Level &l = _mm->resolve(s->levels)[s->levelsUsed];
 
     l.value = value;
     l.capacity = allocated;
     l.label = _mm->base(chars);
+    l.importCapacity = importAllocated;
+    l.importValue = _mm->base(importChars);
```

Comment:
if the user provides a new name, capacity allows us to see if the new name fits, and re-use the same memory without allocating more ... that was the intention with it anyway.